projects/storefrontlib/layout/theme/theme.service.ts

```diff
@@ -60,13 +60,22 @@ export class ThemeService implements OnDestroy {
   }
 
   setTheme(theme: string | undefined): void {
-    if (theme) {
-      const element = this.rootComponent.location.nativeElement;
-      // remove the old theme
-      this.renderer.removeClass(element, this.existingTheme);
-      // add the new theme
-      this.renderer.addClass(element, theme);
-      this.existingTheme = theme;
+    const element = this.rootComponent.location.nativeElement;
+
+    if (this.featureConfigService.isEnabled('useNewSiteThemeSwitcher')) {
+      if (this.existingTheme) {
+        this.renderer.removeClass(element, this.existingTheme);
+      }
+      if (theme) {
+        this.renderer.addClass(element, theme);
+        this.existingTheme = theme;
+      }
```

Comment:
New code accounts for the fact that either `existingTheme` or the new `theme` can be an empty string `''`. But the native DOM API throws an error when we try to add a CSS class `''`.

Moreover, if the old class was non-empty, but the new one is `''`, then we want to just unset the previous one.